pkg/sql/sqlbase/structured.go

```diff
@@ -829,6 +829,83 @@ func (desc *TableDescriptor) AllActiveAndInactiveChecks() []*TableDescriptor_Che
 	return checks
 }
 
+// ReplaceColumnVarsInExprWithDummies replaces the occurrences of column names in an expression with
+// dummies containing their type, so that they may be typechecked. It returns
+// this new expression tree alongside a set containing the ColumnID of each
+// column seen in the expression.
+func (desc *TableDescriptor) ReplaceColumnVarsInExprWithDummies(
+	rootExpr tree.Expr,
+) (tree.Expr, map[ColumnID]struct{}, error) {
+	colIDs := make(map[ColumnID]struct{})
```

Comment:
should we be using `util.FastIntSet` here?

Reply:
Yes, done.